master/freebsd_ci.py

```diff
@@ -0,0 +1,227 @@
+        command=['./.freebsdci.sh', 'cleanup']
+    ),
+
+    steps.Compile(
+        command=['./.freebsdci.sh', 'compile'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+
+    steps.ShellCommand(
+        name='build-stats',
+        command=['./.freebsdci.sh', 'build-state'],
+    ),
+
+    steps.Test(
+        command=['./.freebsdci.sh', 'runtests'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+        timeout=1200,  # 20 min
+    ),
+
+    steps.ShellCommand(
+        name='test embedding',
+        command=['./.freebsdci.sh', 'test-embedding'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+])
+
+###############################################################################
+#  Builders
+###############################################################################
+
+freebsdci_builder_names = {
+    'main': 'freebsdci',
+    'test': 'freebsdci_test',
+}
+
+freebsdci_builders = [
+    util.BuilderConfig(
+        name=freebsdci_builder_names['main'],
+        workernames=['gaebolg', 'abeing'],
+        factory=freebsdci_factory),
+
+    util.BuilderConfig(
+        name=freebsdci_builder_names['test'],
+        workernames=['csisw3', 'fragarach', 'rhongomyniad',
+                     'hrunting', 'balmung'],
+        factory=freebsdci_test_factory),
+]
+
+###############################################################################
+#  Schedulers
+###############################################################################
+
+freebsdci_schedulers = [
+    schedulers.SingleBranchScheduler(
+        name="freebsdci_master",
+        change_filter=util.ChangeFilter(branch='master'),
+        treeStableTimer=60,
+        builderNames=[freebsdci_builder_names['main']]),
+
+    schedulers.ForceScheduler(
+        name="freebsdci_force",
+        builderNames=[freebsdci_builder_names['main']]),
+
+    schedulers.ForceScheduler(
+        name="freebsdci_force_test",
+        builderNames=[freebsdci_builder_names['test']]),
+
+    schedulers.SingleBranchScheduler(
+        name='freebsdci_pull_request',
+        change_filter=util.ChangeFilter(
+            branch_re='^refs/pull/.*',
+            repository_re='https://github.com/[\w]+/julia'),
+        builderNames=[freebsdci_builder_names['main']],
+        treeStableTimer=10,
+    ),
+]
+
+###############################################################################
+#  GitHub Status Reporter
+###############################################################################
+
+freebsdci_report = reporters.GitHubStatusPush(
+    token=FREEBSDCI_OAUTH_TOKEN,
+    context='julia freebsd ci',
+    startDescription='Build started',
+    endDescription='Build done')
```

Comment:
how does this status get connected up to the proper scheduler?

The reason I ask is because on the other schedulers I have a “separated testing” setup working, where the build happens on one machine, and the testing happens on a different machine, so that we catch problems like library paths being embedded within the binary incorrectly, etc.....

Reply:
> how does this status get connected up to the proper scheduler?

ah... I have no idea, too. will survey it later.

Reply:
I found that there is a `tags` argument for filtering builds which should be reported. 
http://docs.buildbot.net/latest/manual/cfg-reporters.html

I will add a tag `freebsdci` for my builds and reporter.

Reply:
I read the source code, seems filtering by tags is not implemented in http reporters
(https://github.com/buildbot/buildbot/blob/master/master/buildbot/reporters/http.py#L75-L78),
but other reporters 
(https://github.com/buildbot/buildbot/blob/master/master/buildbot/reporters/notifier.py#L209-L211).

I will send a PR for http reporters.

Reply:
At this moment, I think we can use the `builders` filter first.

Reply:
This is a good find.  We should create a `tester` tag or something like that and have ALL of our testers report CI status. (but not other jobs)